src/erc20.sol

```diff
@@ -97,4 +116,20 @@ contract ERC20 {
     function move(address src, address dst, uint wad) external {
         transferFrom(src, dst, wad);
     }
+
+    // --- Approve by signature ---
+    function permit(address owner, address spender, uint value, uint deadline, uint8 v, bytes32 r, bytes32 s) external {
```

Comment:
The uniswap implementation is used as reference here, correct?
https://github.com/Uniswap/uniswap-v2-core/blob/master/contracts/UniswapV2ERC20.sol#L81